nautobot/extras/models/metadata.py

```diff
@@ -135,3 +138,70 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
+
+class ObjectMetadata(BaseModel, ChangeLoggedModel):
+    metadata_type = models.ForeignKey(
+        to=MetadataType,
+        on_delete=models.CASCADE,
+        related_name="object_metadatas",
+    )
+    contact = models.ForeignKey(
+        to=Contact,
+        on_delete=models.SET_NULL,
+        related_name="object_metadatas",
+        null=True,
+    )
+    team = models.ForeignKey(
+        to=Team,
+        on_delete=models.SET_NULL,
+        related_name="object_metadatas",
+        null=True,
+    )
+    scoped_fields = JSONArrayField(
+        base_field=models.CharField(max_length=50),
+        help_text="List of scoped fields, only direct fields on the model",
+    )
+    value = models.JSONField(
+        blank=True,
+        null=True,
+        help_text="Relevant data value to an object field or a set of object fields",
+    )
+    assigned_object_type = models.ForeignKey(to=ContentType, on_delete=models.SET_NULL, null=True, related_name="+")
+    assigned_object_id = models.UUIDField(db_index=True)
+    assigned_object = GenericForeignKey(ct_field="assigned_object_type", fk_field="assigned_object_id")
+
+    class Meta:
+        ordering = ["metadata_type"]
+        unique_together = ["metadata_type", "assigned_object_type", "assigned_object_id", "scoped_fields"]
```

Comment:
We'll want clean() to check for intersection of `scoped_fields`, i.e. it shouldn't be possible to have two object-metadata for the same type/assigned-object one with fields [a, b, c] and one with fields [c, d, e]. `unique_together` alone can't catch this case.